--- ../original/hid-sony.c	2020-09-12 20:32:49.164498532 +0400
+++ hid-sony.c	2020-09-12 20:04:26.842113231 +0400
@@ -2466,9 +2466,14 @@
 		 * retrieved with feature report 0x81. The address begins at
 		 * offset 1.
 		 */
-		ret = hid_hw_raw_request(sc->hdev, 0x81, buf,
-				DS4_FEATURE_REPORT_0x81_SIZE, HID_FEATURE_REPORT,
-				HID_REQ_GET_REPORT);
+		ret = DS4_FEATURE_REPORT_0x81_SIZE;
 
 		if (ret != DS4_FEATURE_REPORT_0x81_SIZE) {
 			hid_err(sc->hdev, "failed to retrieve feature report 0x81 with the DualShock 4 MAC address\n");
